src/raftstore/store/peer.rs

```diff
@@ -634,13 +632,21 @@ impl Peer {
             return;
         }
 
-        if !self.raft_group.has_ready() {
+        if self.has_pending_snap() && !self.ready_to_handle_pending_snap() {
+            debug!("{} apply index {} != committed index {}, skip applying snapshot.",
```

Comment:
Then both apply thread and raftstore will try to update apply_state key, which will lead to unpredictable behavior.